src/gdb_main.c

```diff
@@ -423,35 +423,37 @@ static void exec_q_supported(const char *packet, const size_t length)
 
 static void exec_q_memory_map(const char *packet, const size_t length)
 {
+	target_s *target = cur_target;
+
 	(void)packet;
 	(void)length;
 	/* Read target XML memory map */
-	if ((!cur_target) && last_target) {
-		/* Attach to last target if detached. */
-		cur_target = target_attach(last_target, &gdb_controller);
+	if (!target) {
```

Comment:
Not entirely your fault here (the old code needed this fix too), but please drop the braces ({}) as they are unnecessary.